packages/eslint-plugin/rules/i18n-hyphenated-range.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const EN_DASH = '–';
+const HYPHEN_IN_RANGE = /(\d\s+-\s+\d)|(\d-\d)/g;
+
+function replaceHyphenWithEnDash( string ) {
+	if ( string.match( HYPHEN_IN_RANGE ) ) {
+		return string.replace( '-', EN_DASH );
+	}
+}
+
+// see eslint-plugin-wpcalypso.
+function makeFixerFunction( arg ) {
+	return ( fixer ) => {
+		switch ( arg.type ) {
+			case 'TemplateLiteral':
+				return arg.quasis.reduce( ( fixes, quasi ) => {
+					if (
+						'TemplateElement' === quasi.type &&
+						quasi.value.raw.match( HYPHEN_IN_RANGE )
+					) {
+						fixes.push(
+							fixer.replaceTextRange(
+								[ quasi.start, quasi.end ],
+								replaceHyphenWithEnDash( quasi.value.raw )
+							)
+						);
+					}
+					return fixes;
+				}, [] );
+
+			case 'Literal':
+				return [
+					fixer.replaceText(
+						arg,
+						replaceHyphenWithEnDash( arg.raw )
+					),
+				];
+
+			case 'BinaryExpression':
+				return [
+					...makeFixerFunction( arg.left )( fixer ),
+					...makeFixerFunction( arg.right )( fixer ),
+				];
```

Comment:
I verified this works in most cases, however, I spotted a specific case where it produces an unexpected result:
Input: `__( '1-2' + 'Text...' )`
Output: `__( '1–2' + undefined )`
**Expected output:** `__( '1-2' + 'Text…' )`

Similarly, the rule `i18n-ellipsis` has the same issue.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/38225/commits/7e795d0dad70d6472dd8c42e95edf67bf084d5c2

I'm guessing `i18n-ellipsis` can be fixed in a similar manner.